frontend/components/DownloadCsvButton.tsx

```diff
@@ -0,0 +1,31 @@
+import React from "react";
+import { CSVLink } from "react-csv";
+
+interface DownloadCsvButtonProps {
+  data: object[];
```

Comment:
pass the map function as variable in the interface

```
const allAttendeesCsvData = sortedAttendeeEntries.map(({ attendeeName, attendeeDetailsObj, purchaser }) => ({
    "Ticket Count": attendeeDetailsObj.ticketCount,
    "Attendee Name": attendeeName,
    Email: purchaser.email,
    "Phone Number": attendeeDetailsObj.phone ? `'${attendeeDetailsObj.phone}` : "N/A",
  }));
```

e.g., 
type HeaderPair = typeof headers[number]; // { label: "Name", key: "name" } | { label: "Email", key: "email" }

(sortedAttendeeEntries) => HeaderPair[]